godot-core/src/export.rs

```diff
@@ -0,0 +1,161 @@
+    ///     https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#enum-globalscope-propertyhint
+    /// ).
+    fn type_string() -> String;
+}
+
+mod export_impls {
+    use super::*;
+    use crate::builtin::meta::VariantMetadata;
+    use crate::builtin::*;
+
+    macro_rules! impl_export_by_clone {
+        ($Ty:ty => $variant_type:ident) => {
+            impl Export for $Ty {
+                fn export(&self) -> Self {
+                    // If `Self` does not implement `Clone`, this gives a clearer error message
+                    // than simply `self.clone()`.
+                    Clone::clone(self)
+                }
+
+                fn default_export_info() -> ExportInfo {
+                    ExportInfo::new_none(Self::variant_type())
+                }
+            }
+
+            impl TypeString for $Ty {
+                fn type_string() -> String {
+                    format!("{}:", sys::VariantType::$variant_type as i32)
+                }
+            }
+        };
+    }
+
+    impl_export_by_clone!(Aabb => Aabb);
+    impl_export_by_clone!(bool => Bool);
+    impl_export_by_clone!(Basis => Basis);
+    impl_export_by_clone!(Vector2 => Vector2);
+    impl_export_by_clone!(Vector3 => Vector3);
+    impl_export_by_clone!(Vector4 => Vector4);
+    impl_export_by_clone!(Vector2i => Vector2i);
+    impl_export_by_clone!(Vector3i => Vector3i);
+    impl_export_by_clone!(Quaternion => Quaternion);
+    impl_export_by_clone!(Color => Color);
+    impl_export_by_clone!(GodotString => String);
+    impl_export_by_clone!(StringName => StringName);
+    impl_export_by_clone!(NodePath => NodePath);
+    impl_export_by_clone!(PackedByteArray => PackedByteArray);
+    impl_export_by_clone!(PackedInt32Array => PackedInt32Array);
+    impl_export_by_clone!(PackedInt64Array => PackedInt64Array);
+    impl_export_by_clone!(PackedFloat32Array => PackedFloat32Array);
+    impl_export_by_clone!(PackedFloat64Array => PackedFloat64Array);
+    impl_export_by_clone!(PackedStringArray => PackedStringArray);
+    impl_export_by_clone!(PackedVector2Array => PackedVector2Array);
+    impl_export_by_clone!(PackedVector3Array => PackedVector3Array);
+    impl_export_by_clone!(PackedColorArray => PackedColorArray);
+    impl_export_by_clone!(Plane => Plane);
+    impl_export_by_clone!(Projection => Projection);
+    impl_export_by_clone!(Rid => Rid);
+    impl_export_by_clone!(Rect2 => Rect2);
+    impl_export_by_clone!(Rect2i => Rect2i);
+    impl_export_by_clone!(Transform2D => Transform2D);
+    impl_export_by_clone!(Transform3D => Transform3D);
+    impl_export_by_clone!(f64 => Float);
+    impl_export_by_clone!(f32 => Float);
+    impl_export_by_clone!(i64 => Int);
+    impl_export_by_clone!(i32 => Int);
+    impl_export_by_clone!(i16 => Int);
+    impl_export_by_clone!(i8 => Int);
+    impl_export_by_clone!(u32 => Int);
+    impl_export_by_clone!(u16 => Int);
+    impl_export_by_clone!(u8 => Int);
```

Comment:
I know they were already there, but how do these basic types (besides `f64` and `i64`) behave in terms of exporting?

What happens if GDScript assigns a value that doesn't fit into the target type (especially for the integers)?

Reply:
f32 seems to mostly be fine, generally just rounding/becoming inf

the integers however panic in rust. this causes godot to display the weird error of 

![image](https://github.com/godot-rust/gdext/assets/9107253/06be1516-e55e-455c-8b29-1e48c1f7eed5)

though you also get the panic from rust which does state what the issue is:

![image](https://github.com/godot-rust/gdext/assets/9107253/dc2901a4-600c-4f70-93f9-65c801b538c3)

Reply:
Ok, that's at least reasonable behavior. Could you maybe add a short comment above the `impl_export_by_clone!` statements explaining this?